Add Sku resource with array support and CRUD methods

marierinden opened the pull request:
### Description

This pull request introduces the `SkuResource` with complete support for SKU-related operations. Key updates include:

1. **CRUD Operations**:
   - Added methods to perform Create, Read, Update, and Delete operations for SKUs.
   - Implemented validation and data mapping for seamless integration.

2. **Array Field Support in `SkuResponse`**:
   - Updated `products`, `variants`, and `inventories` fields in `SkuResponse` to support arrays.
   - Enhanced data handling for multi-entry SKU attributes.

3. **Request/Response DTOs**:
   - Introduced `SkuRequest`, `SkuResponse`, and `SkuCollectionResponse` for efficient data handling and API communication.

4. **Unit Testing**:
   - Added comprehensive unit tests for `SkuResponse`, `SkuRequest`, and `SkuCollectionResponse` to ensure functionality and reliability.

marierinden committed:
Add Sku resource with request/response DTOs and methods

Introduce SkuResource along with SkuRequest and response DTOs (SkuResponse, SkuCollectionResponse) to manage Sku-related operations. Provides methods for CRUD operations, including find, get, create, update, and delete. Implements validation and mapping for efficient data handling with the client and data mapper.

marierinden committed:
Add support for array fields in SkuResponse

This commit updates `SkuResponse` to support array types for `products`, `variants`, and `inventories` fields. It also introduces corresponding unit tests for SkuResponse, SkuRequest, and SkuCollectionResponse to ensure proper functionality and coverage.

fredrik-tveraaen-apiera commented on src/DTO/Request/Sku/SkuRequest.php:
```diff
@@ -0,0 +1,52 @@
+{
+    public function __construct(
+        #[RequestField('code')]
+        private string $code,
```
Change to nullable

fredrik-tveraaen-apiera commented on src/DTO/Request/Sku/SkuRequest.php:
```diff
@@ -0,0 +1,52 @@
+        #[RequestField('code')]
+        private string $code,
+        #[RequestField('organization')]
+        private string $organization,
```
Remove this as it's an documentation bug

fredrik-tveraaen-apiera commented on src/Resource/SkuResource.php:
```diff
@@ -0,0 +1,167 @@
+ */
+final readonly class SkuResource implements RequestResourceInterface
+{
+    private const string ENDPOINT = '/skus';
```
Add prefix /api/v1/ as skus is not a nested endpoint

marierinden committed:
Refactor SkuRequest and update SkuResource endpoint.

Make `code` optional in `SkuRequest` and remove `organization` field. Adjust tests and serialization accordingly. Update SkuResource endpoint to `/api/v1/skus` for consistency with API versioning.

marierinden committed:
Update `SkuResponse` to make `code` field nullable

Modified the `code` property in `SkuResponse` to be nullable and updated its getter accordingly. Adjusted the corresponding unit test to include the `code` field in the list of nullable fields.

fredrik-tveraaen-apiera committed:
Add support for SKU resource with comprehensive tests

Refactored SKU-related structures, including SkuResponse, and updated tests for consistency with the new SKU resource. Added integration tests for create, update, get, and delete operations. Enhanced `LdType` and `ApieraSdk` to include SKU-specific logic and mappings.

fredrik-tveraaen-apiera committed:
Merge branch 'release/0.3.0' into feature/sku-implementation

fredrik-tveraaen-apiera committed:
Fix whitespace issue in ApieraSdk.php

Removed an unnecessary trailing space to ensure code consistency. This change doesn't affect functionality but improves code readability.

fredrik-tveraaen-apiera reviewed (approved):
:100:
